Add declaration parameters for queue and exchange in rabbitmq event bus

htve opened the pull request:
### Description

Resolves #17401

TODO: Describe what this PR has changed, add screenshot or animated GIF if available, write if it is a breaking change, and how to fix the breaking changes for existing applications if so.

### Checklist

- [ ] I fully tested it as developer / designer and created unit / integration tests
- [ ] I documented it (or no need to document or I will create a separate documentation issue)

### How to test it?

Please describe how this can be tested by the test engineers if it is not already explicit - or remove this section if no need to description.

htve committed:
added QueueType field in RabbitMqEventBusOptions

htve committed:
adding optional arguments when defining rabbitmq queues

htve committed:
added the queuetype argument when declaring rabbitmq queues

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/abpframework/abp?pullRequest=17402) <br/>All committers have signed the CLA.

htve committed:
added  queue and exchange atguments

1. added QueueDurable/QueueExclusive/QueueAutoDelete/QueueArguments
2. added ExchangeDurable/ExchangeAutoDelete/ExchangeArguments

htve committed:
adding optional arguments when defining rabbitmq exchange

htve committed:
Use the corresponding parameters in AbpRabbitMqEventBusOptions when defining queues/switches

htve committed:
Delete unused field QueueType in AbpRabbitMqEventBusOptions

htve committed:
Create RabbitMqOptionalArguments Class

htve committed:
Optimize code format in RabbitMqDistributedEventBus.cs

htve committed:
Use RabbitMQ optional arguments in AbpRabbitMqEventBusOptions

htve committed:
Reme reflections for performance reasons in RabbitMqOptionalArguments

realLiangshiwei commented:
Hi, @htve 

Could you change PR to draft if not done yet?

htve commented:
@realLiangshiwei is completed.

htve committed:
Fix generation errors

htve committed:
Migration Volo.Abp.EventBus.RabbitMQ.AbpRabbitMqEventBusOptions to Volo.Abp.RabbitMQ.AbpRabbitMqExchangeAndQueueOptions

htve commented:
@realLiangshiwei current PR has completed, please review and approve,
LGTM!

htve committed:
Recovery AbpRabbitMqEventBusOptions and inherited from AbpRabbitMqExchangeAndQueueBaseOptions

for backward compatibility, restore

htve committed:
Add comment reference URL to RabbitMQ.OptionalArguments

realLiangshiwei commented on framework/src/Volo.Abp.RabbitMQ/Volo/Abp/RabbitMQ/AbpRabbitMqExchangeAndQueueOptions.cs:
```diff
@@ -0,0 +1,39 @@
+
+    public string? ClientName { get; set; }
+
+    public bool QueueDurable { get; set; }
+
+    public bool QueueExclusive { get; set; } = true;
+    
+    public bool QueueAutoDelete { get; set; } = true;
```
It changes the previous default value

<img width="514" alt="image" src="https://github.com/abpframework/abp/assets/16813853/9e04c9e6-c1c9-4933-986e-39255791fcae">

realLiangshiwei commented on framework/src/Volo.Abp.RabbitMQ/Volo/Abp/RabbitMQ/AbpRabbitMqExchangeAndQueueOptions.cs:
```diff
@@ -0,0 +1,39 @@
+        }
+    }
+
+    public bool ExchangeDurable { get; set; }
```
Should be `true` by default

realLiangshiwei commented on framework/src/Volo.Abp.RabbitMQ/Volo/Abp/RabbitMQ/RabbitMqOptionalArguments.cs:
```diff
@@ -0,0 +1,171 @@
+
+namespace Volo.Abp.RabbitMQ;
+
+public class OptionalArguments
```
It contains the parameters of `queue` and `exchange`, which may be confusing.
I think this class is unnecessary.

Add two dictionary properties, and developers can add additional arguments to the `queue` or `exchange` according to their own needs.
This way is simpler and conforms to the design of RabbitMQ API.  what do you think?
```csharp
public IDictionary<string, object> QueueArguments { get; set; }

public IDictionary<string, object> ExchangeArguments { get; set; }
```

realLiangshiwei commented on framework/src/Volo.Abp.EventBus.RabbitMQ/Volo/Abp/EventBus/RabbitMq/AbpRabbitMqEventBusOptions.cs:
```diff
@@ -2,24 +2,6 @@
 namespace Volo.Abp.EventBus.RabbitMq;
 
-public class AbpRabbitMqEventBusOptions
+public class AbpRabbitMqEventBusOptions : AbpRabbitMqExchangeAndQueueBaseOptions
```
I think we don't need to add `AbpRabbitMqExchangeAndQueueBaseOptions`.

Because it is designed for the EventBus.

realLiangshiwei commented:
Thanks @htve for your contribution, I added some comments

realLiangshiwei commented:
https://github.com/abpframework/abp/pull/18294
